crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -274,9 +274,37 @@ pub fn impl_param_set(_input: TokenStream) -> TokenStream {
     tokens
 }
 
+// replace 'world with 'w and 'state with 's
+fn replace_lifetimes(stream: TokenStream, f: Box<dyn Fn(&str) -> &str>) -> TokenStream {
+    struct LifetimeReplacer {
+        f: Box<dyn Fn(&str) -> &str>,
+    }
+    impl VisitMut for LifetimeReplacer {
+        fn visit_lifetime_mut(&mut self, lifetime: &mut syn::Lifetime) {
+            lifetime.ident = syn::Ident::new(
+                (self.f)(lifetime.ident.to_string().as_str()),
+                lifetime.ident.span(),
+            );
+        }
+    }
+
+    let mut ast = parse_macro_input!(stream as DeriveInput);
+    LifetimeReplacer { f }.visit_derive_input_mut(&mut ast);
+
+    quote! {#ast}.into()
+}
+
 /// Implement `SystemParam` to use a struct as a parameter in a system
 #[proc_macro_derive(SystemParam, attributes(system_param))]
 pub fn derive_system_param(input: TokenStream) -> TokenStream {
+    let input = replace_lifetimes(
+        input,
+        Box::new(|s| match s {
```

Comment:
I feel like we should just not special-case lifetime names at all 🤔 Are we able to do so?

Reply:
I respect your intuition, but i dont understand it. the whole point of that function is to abbreviate lifetimes, since 'world and 'state should be a special exception; replacing other things like variable names or punctuation is really not what were after. 
Honestly id go back to the previous iteration where the values were hardcoded, and declare the function inside the only place where it is used (if rust allows that)
maybe someone with macro experience can actually find a more elegant solution, but idk if there is one

Reply:
Let's do that and solve the issue raised directly.